corehq/apps/case_importer/do_import.py

```diff
@@ -38,8 +38,31 @@
 
 
 def do_import(spreadsheet, config, domain, task=None, record_form_callback=None):
-    importer = _TimedAndThrottledImporter(domain, config, task, record_form_callback)
-    return importer.do_import(spreadsheet)
+    has_domain_column = 'domain' in [c.lower() for c in spreadsheet.get_header_columns()]
+    if has_domain_column:
+        mirror_domains = DomainPermissionsMirror.mirror_domains(domain)
+        sub_domains = set()
+        results = _ImportResults()
+        for row_num, row in enumerate(spreadsheet.iter_row_dicts(), start=1):
+            if row_num == 1:
+                continue
+            sheet_domain = row.get('domain')
+            if sheet_domain != domain and sheet_domain not in mirror_domains:
+                err = exceptions.CaseRowError(column_name='domain')
```

Comment:
@calellowitz Does the cross-domain mobile user upload use `domain` or `project` as the column name? We should be consistent but should also probably not use "domain" in anything user-facing.

Reply:
Confirming the cross-domain user upload does use "domain" so this PR is consistent.